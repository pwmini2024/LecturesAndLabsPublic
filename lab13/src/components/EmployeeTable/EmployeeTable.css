.table-container {
  margin: 20px;
  overflow-x: auto;
}

.employee-table {
  width: 100%;
  border-collapse: collapse;
  background: white;
  box-shadow: 0 1px 3px rgba(0,0,0,0.2);
}

.employee-table th,
.employee-table td {
  padding: 12px;
  text-align: left;
  border-bottom: 1px solid #ddd;
}

.employee-table th {
  background-color: #f4f4f4;
  font-weight: bold;
}

.employee-table tr:hover {
  background-color: #f5f5f5;
}

.delete-btn {
  background-color: #ff4444;
  color: white;
  border: none;
  padding: 5px 10px;
  border-radius: 4px;
  cursor: pointer;
}

.delete-btn:hover {
  background-color: #cc0000;
} 