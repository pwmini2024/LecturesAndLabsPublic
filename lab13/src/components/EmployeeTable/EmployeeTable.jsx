import React from "react";
import "./EmployeeTable.css";

function EmployeeTable({ employees, onDelete }) {
  return (
    <div className="table-container">
      <table className="employee-table">
        <thead>
          <tr>
            <th>Name</th>
            <th>Position</th>
            <th>Department</th>
            <th>Email</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {employees.map((employee) => (
            <tr key={employee.id}>
              <td>{employee.name}</td>
              <td>{employee.position}</td>
              <td>{employee.department}</td>
              <td>{employee.email}</td>
              <td>
                <button
                  className="delete-btn"
                  onClick={() => onDelete(employee.id)}
                  aria-label={`Delete ${employee.name}`}
                >
                  Delete
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default EmployeeTable;
