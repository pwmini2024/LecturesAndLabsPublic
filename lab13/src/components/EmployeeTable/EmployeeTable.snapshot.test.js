import React, { act } from "react";
import { render } from "@testing-library/react";
import EmployeeTable from "./EmployeeTable";

describe("EmployeeTable Snapshots", () => {
  const mockEmployees = [
    {
      id: 1,
      name: "John Doe",
      position: "Developer",
      department: "IT",
      email: "john@example.com",
    },
  ];

  it("should match snapshot with employees", async () => {
    let container;
    await act(async () => {
      const result = render(
        <EmployeeTable employees={mockEmployees} onDelete={() => {}} />
      );
      container = result.container;
    });

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with empty employees list", async () => {
    let container;
    await act(async () => {
      const result = render(
        <EmployeeTable employees={[]} onDelete={() => {}} />
      );
      container = result.container;
    });

    expect(container).toMatchSnapshot();
  });
});
