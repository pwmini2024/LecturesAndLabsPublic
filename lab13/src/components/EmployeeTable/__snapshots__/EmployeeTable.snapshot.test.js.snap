// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeTable Snapshots should match snapshot with employees 1`] = `
<div>
  <div
    class="table-container"
  >
    <table
      class="employee-table"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Position
          </th>
          <th>
            Department
          </th>
          <th>
            Email
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            John Doe
          </td>
          <td>
            Developer
          </td>
          <td>
            IT
          </td>
          <td>
            john@example.com
          </td>
          <td>
            <button
              aria-label="Delete John Doe"
              class="delete-btn"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`EmployeeTable Snapshots should match snapshot with empty employees list 1`] = `
<div>
  <div
    class="table-container"
  />
</div>
`;
